### PR TITLE
Add GH action that will run our tests

### PR DESCRIPTION
**Pull request by kanzelm3**
Run tests on every commit

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -264,15 +264,16 @@ export const TernaryComponent = () => {
...
-const ifBlock = /* tsx */ `
-export const IfBlockComponent = () => {
-  const active = useIsActive();
-  if (active) {
-    return <ActiveComponent />;
-  }
-  return <div />;
-};
-`;
+// TODO: Implement fix for if block conditionals
```
Comment (kanzelm3) [inline test.js:287]: I had a failing test written for if block conditionals, so I just commented it out until we have a fix for it. I suppose I could also just leave it failing as a forcing function for us to fix it though.

File: package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "jest": "^29.3.1",
     "prettier": "^2.7.0",
     "pretty-quick": "^3.1.0",
-    "pro-commit": "^1.2.0"
+    "pro-commit": "^1.2.0",
+    "typescript": "^4.9.4"
```
Comment (Christian0411) [inline package.json:24]: What is typescript used for here?

File: test.js (hunk 1 above)
Comment (amcoder) [reply test.js:287]: I got rid of this too.

**Review by amcoder — APPROVED** [review]
This is awesome!

File: package.json (hunk 2 above)
Comment (Christian0411) [reply package.json:24]: Nevermind I see its required for the action

**Review by Christian0411 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (kanzelm3) [reply package.json:24]: It's apparently a peer dependency of `@typescript-eslint/parser` or something, as when it tried to run our tests previously it failed the it couldn't find the `typescript` module.

**Pull request merged**